Mathlib/Condensed/Equivalence.lean

```diff
@@ -0,0 +1,306 @@
+noncomputable
+def equivalence (A : Type _) [Category.{u+1} A] [HasLimits A] :
+    Sheaf (coherentTopology Stonean) A ≌ Condensed.{u} A :=
+  CoverDense.sheafEquivOfCoverPreservingCoverLifting coverDense coverPreserving coverLifting
+
+end StoneanCompHaus
+
+end Condensed
+
+section ProjectiveProfinite -- This section is PR #6809
+
+namespace Stonean
+
+/-- Every Stonean space is projective in `Profinite` -/
+instance (X : Stonean) : Projective (toProfinite.obj X) where
+  factors := by
+    intro B C φ f _
+    haveI : ExtremallyDisconnected (toProfinite.obj X) := X.extrDisc
+    have hf : f.1.Surjective
+    · rwa [Profinite.epi_iff_surjective] at *
+    obtain ⟨f', h⟩ := CompactT2.ExtremallyDisconnected.projective φ.continuous f.continuous hf
+    use ⟨f', h.left⟩
+    ext
+    exact congr_fun h.right _
+
+end Stonean
+
+namespace Profinite
+
+instance {X Y : Profinite} (f : X ⟶ Y) [Epi f] : @Epi CompHaus _ _ _ f := by
+  rwa [CompHaus.epi_iff_surjective, ← epi_iff_surjective]
+
+instance {X Y : Profinite} (f : X ⟶ Y) [@Epi CompHaus _ _ _ f] : Epi f := by
+  rwa [epi_iff_surjective, ← CompHaus.epi_iff_surjective]
+
+/-- If `X` is compact Hausdorff, `presentation X` is an extremally disconnected space
+  equipped with an epimorphism down to `X`. It is a "constructive" witness to the
+  fact that `CompHaus` has enough projectives.  -/
+noncomputable
+def presentation (X : Profinite) : Stonean where
+  compHaus := X.toCompHaus.projectivePresentation.p
+  extrDisc := X.toCompHaus.presentation.extrDisc
+
+/-- The morphism from `presentation X` to `X`. -/
+noncomputable
+def presentation.π (X : Profinite) : Stonean.toProfinite.obj X.presentation ⟶ X :=
+  X.toCompHaus.projectivePresentation.f
+
+/-- The morphism from `presentation X` to `X` is an epimorphism. -/
+noncomputable
+instance presentation.epi_π (X : Profinite) : Epi (π X) := by
+  have := X.toCompHaus.projectivePresentation.epi
+  rw [CompHaus.epi_iff_surjective] at this
+  rw [Profinite.epi_iff_surjective]
+  exact this
+
+/--
+```
+               X
+               |
+              (f)
+               |
+               \/
+  Z ---(e)---> Y
+```
+If `Z` is extremally disconnected, X, Y are profinite, if `f : X ⟶ Y` is an epi and
+`e : Z ⟶ Y` is arbitrary, then `lift e f` is a fixed (but arbitrary) lift of `e` to a morphism
+`Z ⟶ X`. It exists because `Z` is a projective object in `CompHaus`.
+-/
+noncomputable
+def lift {X Y : Profinite} {Z : Stonean} (e : Stonean.toProfinite.obj Z ⟶ Y) (f : X ⟶ Y) [Epi f] :
+    Stonean.toProfinite.obj Z ⟶ X := CompHaus.lift e f
+
+@[simp, reassoc]
+lemma lift_lifts {X Y : Profinite} {Z : Stonean} (e : Stonean.toProfinite.obj Z ⟶ Y) (f : X ⟶ Y)
+    [Epi f] : lift e f ≫ f = e := CompHaus.lift_lifts _ _
+
+lemma projective_of_extrDisc {X : Profinite} (hX : ExtremallyDisconnected X) :
+    Projective X:= by
+  let X' : Stonean := ⟨X.toCompHaus⟩
+  show Projective (Stonean.toProfinite.obj X')
+  exact inferInstance
+
+end Profinite
```

Comment:
This stuff is now all part of `Topology/Category/Stonean/Basic`.

Reply:
Oh sorry I didn't merge master after #6809 was merged.